File: Programmers/[12905] 가장 큰 정사각형 찾/ONE/12905.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] board) {
        int n = board.length;
        int m = board[0].length;

        for (int row = 1; row < n; row++) {
            for (int col = 1; col < m; col++) {
                if (board[row][col] == 1) {
                    board[row][col] = Math.min(board[row - 1][col - 1],
                            Math.min(board[row][col - 1], board[row - 1][col])) + 1;
                }
            }
        }

        int max = Arrays.stream(board)
                .flatMapToInt(Arrays::stream)
                .max()
                .getAsInt();

        return max * max;
    }
}

File: Programmers/[12905] 가장 큰 정사각형 찾기/KingDonggyu/12905.js
function checkIsOutOfBounds(x, y, n, m) {
  return x < 0 || y < 0;
}

function solution(board) {
  let answer = 0;
  const N = board.length;
  const M = board[0].length;

  board.forEach((row, x) => {
    row.forEach((_, y) => {
      if (
        !board[x][y] ||
        checkIsOutOfBounds(x - 1, y - 1, N, M) ||
        checkIsOutOfBounds(x - 1, y, N, M) ||
        checkIsOutOfBounds(x, y - 1, N, M)
      ) {
        return;
      }

      board[x][y] =
        Math.min(board[x - 1][y - 1], board[x - 1][y], board[x][y - 1]) + 1;

      answer = Math.max(answer, board[x][y]);
    });
  });

  return answer * answer;
}
